src/test/java/com/github/fppt/jedismock/datastructures/streams/StreamIdTests.java

```diff
@@ -0,0 +1,259 @@
+        StreamId zero = new StreamId(0, 0);
+        assertThrows(
+                WrongStreamKeyException.class,
+                zero::compareToZero,
+                "ERR The ID specified in XADD must be greater than 0-0"
+        );
+    }
+
+    @Test
+    void zeroComparisonWithPositiveKeysTest() {
+        StreamId other = new StreamId(0, 1);
+        assertDoesNotThrow(() -> assertSame(other, other.compareToZero()));
+
+        StreamId newOther = new StreamId(1, 0);
+        assertDoesNotThrow(() -> assertSame(newOther, newOther.compareToZero()));
+
+        StreamId newestOther = new StreamId(1, 1);
+        assertDoesNotThrow(() -> assertSame(newestOther, newestOther.compareToZero()));
+    }
+
+    @Test
+    void zeroComparisonWithNegativeKeysTest() {
+        StreamId other = new StreamId(0, -1);
+        assertDoesNotThrow(() -> assertSame(other, other.compareToZero()));
+
+        StreamId newOther = new StreamId(-1, 0);
+        assertDoesNotThrow(() -> assertSame(newOther, newOther.compareToZero()));
+
+        StreamId newestOther = new StreamId(-1, -1);
+        assertDoesNotThrow(() -> assertSame(newestOther, newestOther.compareToZero()));
+    }
+
+    @Test
+    void incrementMaxKeyTest() {
+        StreamId maxKey = new StreamId(-1, -1);
+        assertThrows(
+                WrongStreamKeyException.class,
+                maxKey::increment,
+                "ERR invalid start ID for the interval"
+        );
+    }
+
+    @Test
+    void incrementTest() {
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(1234567890, 1234567891),
+                new StreamId(1234567890, 1234567890).increment()
+        ));
+
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(1, 0),
+                new StreamId(0, -1).increment()
+        ));
+
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(-1234567891, -1234567890),
+                new StreamId(-1234567891, -1234567891).increment()
+        ));
+    }
+
+    @Test
+    void decrementMinKeyTest() {
+        StreamId minKey = new StreamId(0, 0);
+        assertThrows(
+                WrongStreamKeyException.class,
+                minKey::decrement,
+                "ERR invalid end ID for the interval"
+        );
+    }
+
+    @Test
+    void decrementTest() {
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(1234567890, 1234567890),
+                new StreamId(1234567890, 1234567891).decrement()
+        ));
+
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(0, -1),
+                new StreamId(1, 0).decrement()
+        ));
+
+        assertDoesNotThrow(() -> assertEquals(
+                new StreamId(-1234567891, -1234567891),
+                new StreamId(-1234567891, -1234567890).decrement()
+        ));
+    }
+
+    @Test
+    void incrementDecrementFuzzingTest() {
+        for (int i = 0; i < 1000; ++i) {
```

Comment:
Please do not write such tests. They are not real fuzzing tests, and you should not copy the approaches from TCL native tests in Redis (they are restricted there by the available tooling and are quite foolish "poor man's fuzzing").

Real fuzzing is more complex than this, it's a combination of well-thought selection of "bad", "corner-case" examples with tooling that makes any failure reproducible (if something fails, we must be able to know which example failed).

Fuzzing for Java is provided by [jqwik](https://jqwik.net/), but I'm in doubt whether we should pull  `jqwik` into this project. Just a parameterized test with a couple of examples will cover everything and will not bring any additional value.

Please, do not write unit tests like this, use parameterized tests instead for 2-3 examples including corner cases